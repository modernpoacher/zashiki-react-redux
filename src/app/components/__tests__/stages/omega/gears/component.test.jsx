import React from 'react'
import PropTypes from 'prop-types'

import {
  toSnapshot
} from 'react-component-snapshot'

import '@testing-library/jest-dom'

import {
  render
} from '@testing-library/react'

import Gears from '#zashiki-react-redux/app/components/stages/omega/gears/component'

jest.mock('shinkansen-engine/components/signals', () => ({
  OMEGA_PATTERN: 'MOCK OMEGA PATTERN'
}))

/**
 *  @param {{ to: string | { pathname: string }, children: React.ReactNode | React.ReactNode[] }} props
 *  @returns {React.JSX.Element}
 */
function MockLink ({ to, children }) {
  if (typeof to === 'string') {
    return (
      <a href={to} className='mock-link'>
        {children}
      </a>
    )
  }

  const {
    pathname
  } = to

  return (
    <a href={pathname} className='mock-link'>
      {children}
    </a>
  )
}

MockLink.propTypes = {
  to: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.shape({})
  ]),
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(
      PropTypes.node
    )
  ])
}

jest.mock('react-router', () => {
  return {
    __esModule: true,
    Link: MockLink
  }
})

describe('#zashiki-react-redux/app/components/stages/omega/gears/component', () => {
  describe('Always', () => {
    it('renders', () => {
      expect(toSnapshot(render(
        <Gears />
      )))
        .toMatchSnapshot()
    })
  })

  it('renders `reverse`', () => {
    expect(toSnapshot(render(
      <Gears
        reverse={{ alpha: 'MOCK ALPHA', omega: 'MOCK OMEGA' }}
      />
    )))
      .toMatchSnapshot()
  })

  it('renders `forward`', () => {
    expect(toSnapshot(render(
      <Gears
        forward={{ alpha: 'MOCK ALPHA', omega: 'MOCK OMEGA' }}
      />
    )))
      .toMatchSnapshot()
  })

  it('renders `reverse` and `forward`', () => {
    expect(toSnapshot(render(
      <Gears
        reverse={{ alpha: 'MOCK ALPHA', omega: 'MOCK OMEGA' }}
        forward={{ alpha: 'MOCK ALPHA', omega: 'MOCK OMEGA' }}
      />
    )))
      .toMatchSnapshot()
  })
})
