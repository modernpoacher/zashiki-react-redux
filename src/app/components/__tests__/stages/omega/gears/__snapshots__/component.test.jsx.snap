// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@modernpoacher/zashiki-react-redux/app/components/stages/omega/gears/component Always renders 1`] = `null`;

exports[`@modernpoacher/zashiki-react-redux/app/components/stages/omega/gears/component renders \`forward\` 1`] = `null`;

exports[`@modernpoacher/zashiki-react-redux/app/components/stages/omega/gears/component renders \`reverse\` 1`] = `null`;

exports[`@modernpoacher/zashiki-react-redux/app/components/stages/omega/gears/component renders \`reverse\` and \`forward\` 1`] = `null`;
